rasa/cli/run.py

```diff
@@ -77,26 +77,49 @@ def run(args: argparse.Namespace):
 
     args.model = _validate_model_path(args.model, "model", DEFAULT_MODELS_PATH)
 
+    args.endpoints = get_validated_path(
+        args.endpoints, "endpoints", DEFAULT_ENDPOINTS_PATH, True
+    )
+    args.credentials = get_validated_path(
+        args.credentials, "credentials", DEFAULT_CREDENTIALS_PATH, True
+    )
+
     if not args.enable_api:
-        # if the API is enabled you can start without a model as you can train a
-        # model via the API once the server is up and running
+        # if the API is not enable you cannot start without a model
+        # make sure either a model server, a remote storage, or a local model is
+        # configured
+        import sys
         from rasa.model import get_model
+        from rasa.core.utils import AvailableEndpoints
+
+        try:
+            endpoints = AvailableEndpoints.read_endpoints(args.endpoints)
+            model_server = endpoints.model if endpoints and endpoints.model else None
+            model_server_set = model_server is not None
+        except ValueError:
+            model_server_set = False
 
+        remote_storage_set = args.remote_storage is not None
+
+        local_model_set = True
         try:
             get_model(args.model)
         except ModelNotFound:
+            local_model_set = False
```

Comment:
Maybe we only do this try/except if model_server_set and remote_storage_set are false? Since we unnecessarily create dirs and confusing debug log messages (using default location, when in the end that default location won' t be used) in here https://github.com/RasaHQ/rasa/pull/3989/files#diff-9bfad703ea56804fd2844423a92289e9R58 
(so we would move L78 into the try/except too)

Reply:
We cannot move the `_validate_model_path` inside the try/except as it is also needed if `--enable-api` is set.

Reply:
Okay. I do think we should fix how that function works so that it's not unnecessarily creating folders, but that's a bit oos from this if it's used in other places so I'll create a new issue.

Reply:
or maybe we just use it in L88 and 120 since these are the only 2 run options that actually use the model argument